src/commitlint/cli.py

```diff
@@ -119,36 +126,46 @@ def _get_commit_message_from_file(filepath: str) -> str:
         return commit_message
 
 
-def _handle_commit_message(commit_message: str, skip_detail: bool) -> None:
+def _handle_commit_message(
+    commit_message: str, skip_detail: bool, quiet: bool = False
+) -> None:
     """
     Handles a single commit message, checks its validity, and prints the result.
 
     Args:
         commit_message (str): The commit message to be handled.
         skip_detail (bool): Whether to skip the detailed error linting.
+        quiet (bool): Whether to ignore stout and stderr
 
     Raises:
         SystemExit: If the commit message is invalid.
     """
     success, errors = lint_commit_message(commit_message, skip_detail=skip_detail)
 
+    if quiet and not success:
```

Comment:
suggestion: There are a lot of if statements that increase the cyclomatic complexity of this block. Also, some of these if statements can be combined. For eg: line 145 and line 152 both have `not success` check, they can be combined.

Reply:
@sugat009, I had the same query. We could have used the early return/exit technique. However, we're mocking sys.exit in our test code, causing the tests to fail.

Reply:
In that case, we should be changing the tests as well.

Reply:
Yes, I was considering creating a new issue for refactoring the code and tests, as we're currently mocking sys.exit in all of our existing tests.

Reply:
Both are good options. Let's create an issue for refactoring the code and tests but also not leave this feature's refactoring to another ticket.

Reply:
if success and quiet:
        return
    
    if success and not quiet:
        sys.stdout.write(f"{VALIDATION_SUCCESSFUL}\n")
        return

    if not quiet:
        _show_errors(commit_message, errors, skip_detail=skip_detail)
    
    sys.exit(1)
 
 Will this be good enough to clean up the code ?  @aj3sh  @sugat009

Reply:
LGTM. Thanks for the changes @golesuman .

Reply:
Can we think similar kind of refactoring for the `_handle_multiple_commit_messages` function?

```python
has_error = False
for commit_message in commit_messages:
    success, errors = lint_commit_message(commit_message, skip_detail=skip_detail)
    if success:
        continue

    has_error = True
    if not quiet:
        _show_errors(commit_message, errors, skip_detail=skip_detail)
        sys.stderr.write("\n")
```
Let's check if this works.

Reply:
FYI. I force pushed my local changes so I accidentally removed README but I again rebased from main and pushed it so its there now. I also checked your code and it works. @aj3sh